src/main/java/com/sap/cloudfoundry/client/facade/adapters/RawCloudRoute.java

```diff
@@ -1,36 +1,30 @@
 package com.sap.cloudfoundry.client.facade.adapters;
 
-import java.util.List;
-
-import org.cloudfoundry.client.v2.Resource;
-import org.cloudfoundry.client.v2.routemappings.RouteMappingEntity;
-import org.cloudfoundry.client.v2.routes.RouteEntity;
+import org.cloudfoundry.client.v3.routes.RouteResource;
 import org.immutables.value.Value;
 
 import com.sap.cloudfoundry.client.facade.domain.CloudDomain;
 import com.sap.cloudfoundry.client.facade.domain.CloudRoute;
-import com.sap.cloudfoundry.client.facade.domain.Derivable;
 import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudRoute;
+import reactor.util.function.Tuple2;
 
 @Value.Immutable
 public abstract class RawCloudRoute extends RawCloudEntity<CloudRoute> {
 
-    public abstract Resource<RouteEntity> getResource();
-
-    public abstract List<Resource<RouteMappingEntity>> getRouteMappingResources();
-
-    public abstract Derivable<CloudDomain> getDomain();
+    @Value.Parameter
+    public abstract Tuple2<RouteResource, CloudDomain> getRouteAndDomain();
 
     @Override
     public CloudRoute derive() {
-        Resource<RouteEntity> resource = getResource();
-        RouteEntity entity = resource.getEntity();
+        Tuple2<RouteResource, CloudDomain> tuple = getRouteAndDomain();
+        RouteResource route = tuple.getT1();
+        CloudDomain domain = tuple.getT2();
         return ImmutableCloudRoute.builder()
-                                  .metadata(parseResourceMetadata(resource))
-                                  .appsUsingRoute(getRouteMappingResources().size())
-                                  .host(entity.getHost())
-                                  .domain(getDomain().derive())
-                                  .hasServiceUsingRoute(entity.getServiceInstanceId() != null)
```

Comment:
This is a bit tricky because the `Destination` sub-object of a route contains only references to apps. Even though user-provided services _can_ have an optional url value (although it's still an experimental feature), they aren't referenceable as of CAPI version 3.92.0.